site/js/components/ee.js

```diff
@@ -0,0 +1,230 @@
+  }
+
+  /**
+   * Sets the current prompt value.
+   * @param {number} value - the current prompt integer value
+   * @return {void}
+   */
+  _setCurrent(value) {
+    localStorage.setItem('chromeos-m100-current-prompt', value);
+    this._current = value;
+  }
+
+  /**
+   * Clear the current prompt value.
+   * @return {void}
+   */
+  _clearCurrent() {
+    localStorage.removeItem('chromeos-m100-current-prompt');
+    this._current = 0;
+  }
+
+  /**
+   * Gets the current quiz start value from local storage.
+   * @return {number} - current quiz start value
+   */
+  _isStarted() {
+    return !!localStorage.getItem('chromeos-m100-quiz-start');
+  }
+
+  /**
+   * Sets the current quiz start value.
+   * @param {number} value - the current quiz start integer value
+   * @return {void}
+   */
+  _setStarted(value) {
+    localStorage.setItem('chromeos-m100-quiz-start', value);
+    this._started = value;
+  }
+
+  /**
+   * Clear the current quiz start value.
+   * @return {void}
+   */
+  _clearStarted() {
+    localStorage.removeItem('chromeos-m100-quiz-start');
+    this._started = false;
+  }
+
+  /**
+   * Prompt the question to the user
+   * @param {number} prompt
+   * @param {boolean} ret - return text (true) vs prompt text (false|default)
+   * @return {string|void}
+   */
+  _promptQuestion(prompt = 0, ret = false) {
+    if (prompt < this._prompts.length && this._prompts[prompt].question.length > 0) {
+      if (ret) {
+        return this._prompts[prompt].question;
+      }
+      console.info('Question #%s: %c%s', prompt + 1, 'font-weight: bold;', this._prompts[prompt].question);
+    } else if ((prompt = this._prompts.length)) {
+      this._promptReward();
+    }
+  }
+
+  /**
+   * Prompt the hint to the user
+   * @param {string} hint
+   * @return {void}
+   */
+  _promptHint(hint) {
+    if (hint) {
+      console.info('Hint: %c%s', 'font-style: italic;', hint);
+    }
+  }
+
+  /**
+   * Prompt reward
+   * @return {void}
+   */
+  _promptReward() {
+    console.info('%c%s', 'font-weight: bold; font-size: 1.25em;', this._reward.headline);
+    for (const iterator of this._reward.body) {
+      console.info(iterator);
+    }
+  }
+
+  /**
+   * Check the answer
+   * @param {number} prompt
+   * @param {string} value
+   * @return {void}
+   */
+  _checkAnswer(prompt, value) {
+    const prompts = this._prompts[prompt];
+    if (value.toString().toLowerCase() == prompts.answer.toString().toLowerCase()) {
+      // Correct
+      console.info('You answered: %c%s', 'font-weight: bold; font-style: italic; background: #0d652d; color: white; padding: .25em;', value.toString());
+      console.info(prompts.responseCorrect);
+      this._setCurrent(prompt + 1);
+      if (this._current < this._prompts.length) {
+        this._promptQuestion(this._current);
+      } else if ((this._current = this._prompts.length)) {
+        this._promptReward();
+      }
```

Comment:
Doesn't `promptQuestion` have the `promptReward` logic included in it?

Reply:
this has been re-refined.